### DISCUSSION TITLE
Why Is `cryptography` built from source?

### DISCUSSION THREAD

**Discussion by stumpylog** [caronc/apprise-api#148] [Q&A]
I was looking at some possible improvements to the Dockerfile and curious why `cryptography` is build from source?  It provides wheels for both the build platforms (x86_64 and ARM64), so why does the build not use those?

**Comment by stumpylog** [comment]
Even with building from source, I can shrink the image significantly:
```shell
me@server[~/apprise-api]: docker image ls
REPOSITORY       TAG       IMAGE ID       CREATED         SIZE
apprise          test      1c2c32027066   2 minutes ago   234MB
caronc/apprise   latest    8fc9dd34b1f6   4 weeks ago     2.03GB
```

**Reply by caronc** [reply]
Thank you for doing this and sharing your work!
